Makefile.refactor.make

```diff
@@ -105,13 +109,185 @@ protoc-clean:
 # CLEAN
 # =============================================================================
 .PHONY: clean
-clean: protoc-clean
+clean: bin-clean protoc-clean
+
+# =============================================================================
+# CHECK
+# formatting, linting and vetting checks on source files
+# =============================================================================
+.PHONY: check fmt
+# ignore generated files when formatting/linting/vetting
+CHECKSRC := $(shell find . -type f -name '*.go' -not -name '*.pb.go' -not -name '*.pb.gw.go' $(EXCLUDE_DIRS_FILTER))
+
+check:
+	@test -z $(shell $(DOCKER_RUN_CMD) -v $${PWD}:/go/src/$(REPO) -w /go/src/$(REPO) $(GOTOOLS) gofmt -l ${CHECKSRC} | tee /dev/stderr) || echo "[WARN] Fix formatting issues with 'make fmt'"
+	@$(DOCKER_RUN_CMD) -v $${PWD}:/go/src/$(REPO) -w /go/src/$(REPO) $(GOTOOLS) bash -c 'for p in $$(go list ./... | grep -v -e /vendor/ -e /images/ -e /.git/); do golint $${p} | sed "/pb\.\(gw\.\)*go/d"; done'
+	@$(DOCKER_RUN_CMD) -v $${PWD}:/go/src/$(REPO) -w /go/src/$(REPO) $(GOTOOLS) go tool vet ${CHECKSRC}
+
+# format and simplify if possible (https://golang.org/cmd/gofmt/#hdr-The_simplify_command)
+fmt:
+	@$(DOCKER_RUN_CMD) -v $${PWD}:/go/src/$(REPO) -w /go/src/$(REPO) $(GOTOOLS) gofmt -s -l -w $(CHECKSRC)
 
 # =============================================================================
 # BUILD
 # =============================================================================
-build: $(PROTOTARGETS)
-	@echo To be implemented...
+
```

Comment:
We're stepping on each other's toes here. I showed you earlier today what I was doing with the build and have just been fine-tuning it. I asked you to hold off until I was ready to hand over to you to tackle everything beyond the cli. I've got the the parts of the workflow handled from glide, protoc, through building the cli fully containerized as well, I just didn't want to submit the PR until after a few more tweaks. You're PR is stepping on my toes and going to make it unnecessarily hard to resolve things that I've already said I was working on.